rotkehlchen/rotkehlchen.py

```diff
@@ -746,27 +747,51 @@ def query_balances(
                 if len(loopring_balances) != 0:
                     balances[str(Location.LOOPRING)] = loopring_balances
 
-        # retrieve uniswap v3 lps
+        # retrieve uniswap v3 & v2 lps
         uniswap = self.chain_manager.get_module('uniswap')
-        uniswap_balances = None
+        uniswap_v3_balances = None
         if uniswap is not None:
             try:
-                uniswap_balances = uniswap.get_v3_balances(
+                uniswap_v3_balances = uniswap.get_v3_balances(
                     addresses=self.chain_manager.queried_addresses_for_module('uniswap'),
                 )
             except RemoteError as e:
                 log.error(
                     f'At balance snapshot Uniswap V3 balances query failed due to {str(e)}. Error '
                     f'is ignored and balance snapshot will still be saved.',
                 )
+            try:
+                uniswap_v2_balances = uniswap.get_balances(
+                    addresses=self.chain_manager.queried_addresses_for_module('uniswap'),
+                )
+            except RemoteError as e:
+                log.error(
+                    f'At balance snapshot Uniswap V2 balances query failed due to {str(e)}. Error '
+                    f'is ignored and balance snapshot will still be saved.',
+                )
+            else:
+                if str(Location.BLOCKCHAIN) not in balances:
+                    balances[str(Location.BLOCKCHAIN)] = {}
+
+                for uniswap_v2_lps in uniswap_v2_balances.values():
+                    for lp in uniswap_v2_lps:
+                        lp_token = get_or_create_ethereum_token(
```

Comment:
This should not be here, right? It should be inside `uniswap.get_balances()`

Reply:
Yeah.